lib/livebook_web/controllers/auth_controller.ex

```diff
@@ -0,0 +1,20 @@
+defmodule LivebookWeb.AuthController do
+  use LivebookWeb, :controller
+
+  alias LivebookWeb.Helpers
+
+  def index(conn, assigns) do
+    conn
+    |> put_view(LivebookWeb.ErrorView)
+    |> render("401_secret.html", Map.put(assigns, :type, :password))
+  end
+
+  def authenticate(conn, %{"password" => password}) do
+    password = :crypto.hash(:sha256, password) |> Base.encode16()
+    cookie_key = Helpers.auth_cookie_key(conn, :password)
+
+    conn
+    |> put_resp_cookie(cookie_key, password, Helpers.auth_cookie_opts())
+    |> redirect(to: "/")
```

Comment:
Instead of using a cookie, let's compare the password here and store a session key. It is probably preferred to not store the password anywhere on the client. If the password is invalid, we should show a decent error message back in the form. :)

Reply:
What if the password changes? i.e. we restart the deployed server with a new password set. Then users would still be authenticated until someone changes secret key base. Also it's not the password we are storing but its hash instead, is this sill a concern?

Reply:
I see. Ok please ignore me. :)